docs: clarify cross-partition traffic reqs (#17232)
hashicorp/consul: website/content/docs/connect/gateways/mesh-gateway/service-to-service-traffic-partitions.mdx

* Specify `mesh-gateway` in the `kind` field to register the gateway with Consul.
* Configure the `proxy.upstreams` parameters to route traffic to the correct service, namespace, and partition. Refer to the [`upstreams` documentation](/consul/docs/connect/registration/service-registration#upstream-configuration-reference) for details. The service `proxy.upstreams.destination_name` is always required. The `proxy.upstreams.destination_partition` must be configured to enable cross-partition traffic. The `proxy.upstreams.destination_namespace` configuration is only necessary if the destination service is in a different namespace.

> * Configure the `exported-services` configuration entry to make services from one admin partition available to one or more other datacenter-local partitions. Refer to the [Exported Services documentation](/consul/docs/connect/config-entries/exported-services) for more details.

Reviewer: ```suggestion
* Configure the `exported-services` configuration entry to make services in one admin partition available to other datacenters and local partitions. Refer to the [Exported Services documentation](/consul/docs/connect/config-entries/exported-services) for more details.
```
Ooohh, your writing "datacenter-local" instead of "datacenter/local" for style guide purposes, correct?

Author: The intent of "datacenter-local" is to mean "other partitions in the same datacenter as me". "datacenter" is meant as a qualifier of what is meant by "local".

I'm not sure how else to convey this concisely. Perhaps "datacenter-local" is a term we could define in the glossary (and link to)? Also open to other suggestions. Without such a term, this sentence would look something like:

```suggestion
* Configure the `exported-services` configuration entry to make services from one admin partition available to one or more other partitions in the same datacenter. Refer to the [Exported Services documentation](/consul/docs/connect/config-entries/exported-services) for more details.
```

  * A service must be exported to a consuming partition for the consuming partition to have access to that service.

>   * The mesh gateway service must also be exported to each consuming partition. Otherwise, the consuming partition will not be able to access other services through the mesh gateway.

Reviewer: ```suggestion
  * The mesh gateway service must also be exported to each consuming partition. Otherwise, the consuming partition cannot access other services through the mesh gateway.
```
